### DISCUSSION TITLE
Combining getServerSideProps and getStaticPaths

### DISCUSSION THREAD

**Discussion by guykogel** [vercel/next.js#10934] [Help]
Most of the elements on my page are static and only one of them comes from the server? Will I need to give up on SSG because of that one element?
Is it possible to combine both SSG and SSR?
I need the whole page to be pre-rendered

**Comment by sandys** [comment]
hi,
im looking for the same thing. in fact, it would be great to have an example that combines getServerSideProps, getStaticProps, getStaticPaths in one example code

**Comment by timneutkens** [comment]
In that case you'd fetch the dynamic data client-side, not during generation.

**Reply by guykogel** [reply]
@timneutkens but then that data will not pre-rendered, no?

**Reply by timneutkens** [reply]
Correct, but if your data is not dynamic (updates more than once per second) you can CDN cache the response.

**Reply by guykogel** [reply]
The data updates once a day. I have to have that data in the source code, so it must be pre-rendered.  From what I understand, the only way to get the data in the source code is using getServerSideProps. Am I missing something? @timneutkens

**Reply by guykogel** [reply]
@timneutkens is it possible to fetch data from CDN live (not in build) and pre-render it using getStaticPaths? Is that what you meant?

**Reply by valse** [reply]
But the `fallback` option of `getStaticPaths` is made to solve this question right?

I think that that was the real powerful feature of the new next.js version: you could prerender some pages and let the others to be "dynamic prerendered" on first page visit.

The only doubt about it is what happen for the SEO indexing: those "forgotten" pages are empty on first request and then served statically... it's also true that the crawlers scan a page a lot of time but I don't know how they can react about this changes.

The best thing will be dynamic render the page on fallback instead of show a Loading component... and let static generation on background like now ;)

**Reply by timneutkens** [reply]
> The only doubt about it is what happen for the SEO indexing: those "forgotten" pages are empty on first request and then served statically... it's also true that the crawlers scan a page a lot of time but I don't know how they can react about this changes.

> The best thing will be dynamic render the page on fallback instead of show a Loading component... and let static generation on background like now ;)

With regards to SEO the current approach is totally fine, Google does a 2 pass render, the second pass is in a real browser instance and will render the page correctly.

Eventually we might bypass the fallback loading page for bots, but initially we didn't to ensure user vs bot is the same.

**Reply by valse** [reply]
I didn't know that Google does 2 pass... so it's ok with `fallback`: and we're talking about the worst case where the crawler is the first "user" but I saw also that thanks to the page preloading the web worker could be the first one too... very powerful!

**Reply by guykogel** [reply]
@timneutkens 
Usually there is a big delay between the passes, so it is not the ideal solution for sites with dynamic content.
Will the support for bots will come out in the near future?
Thanks

**Reply by timneutkens** [reply]
> Usually there is a big delay between the passes, so it is not the ideal solution for sites with dynamic content.

This is no longer true, 2 years ago it would take up to 2 weeks for the second pass. However since last year it takes a few minutes after the first crawl.

**Reply by guykogel** [reply]
@timneutkens in our case it is crucial. We need the bot to get the updated version first shot. Is the solution for bots will come anytime soon?
Thanks

**Reply by Timer** [reply]
@guykogel performing blocking renders for crawlers (or other bots) is on the future roadmap!
If you or your company has exceptional needs, feel free to reach out to enterprise@vercel.com!
